receiver.go

```diff
@@ -154,11 +155,14 @@ func (r *receiver) receiveDatagram(l int) (int, *net.UDPAddr, error) {
 		if err != nil {
 			return 0, nil, err
 		}
-		// TODO: compare addr here?
+		if !addr.IP.Equal(r.addr.IP) || (r.tid != 0 && addr.Port != r.tid) {
+			return 0, nil, err
```

Comment:
Probably instead of `return 0, nil, err` should be `continue`?

Because `continue` just does what kernel otherwise would do on connected socket: just drop the frame.
In the other hand `return 0, nil, err` is in fact `return 0, nil, nil` here.